### PR TITLE
chore: benchee ingest benchmarks

### PR DESCRIPTION
**Pull request by Ziinc**
This PR adds in benchee as well as some rudimentary ingestion benchmarks, for perf improvements.

benchmarks are tagged with the `:benchmark` test tag.

Inputs generated are meant for ballpark ingest shapes, not to be perfectly accurate.

Sample:
```
Running tests...

01:30:48.421 [info] Migrations already up
Excluding tags: [:test, {:integration, true}, {:failing, true}]
Including tags: [:benchmark]

Operating System: macOS
CPU Information: Apple M1 Pro
Number of Available Cores: 10
Available memory: 32 GB
Elixir 1.16.0
Erlang 26.2.1
JIT enabled: true

Benchmark suite executing with the following configuration:
warmup: 1 s
time: 2 s
memory time: 1 s
reduction time: 1 s
parallel: 1
inputs: none specified
Estimated total run time: 15 s

Benchmarking batched ...
Benchmarking big_single ...
Benchmarking single ...
Calculating statistics...
Formatting results...

Name                 ips        average  deviation         median         99th %
single           3590.39        0.28 ms   ±145.76%        0.21 ms        2.84 ms
big_single        770.43        1.30 ms    ±81.36%        1.25 ms        1.70 ms
batched           106.77        9.37 ms    ±24.94%        9.41 ms       20.50 ms

Comparison: 
single           3590.39
big_single        770.43 - 4.66x slower +1.02 ms
batched           106.77 - 33.63x slower +9.09 ms

Memory usage statistics:

Name               average  deviation         median         99th %
single           0.0999 MB     ±0.80%      0.0993 MB       0.101 MB
big_single         2.01 MB     ±0.27%        2.01 MB        2.01 MB
batched           10.03 MB     ±0.85%       10.01 MB       10.53 MB

Comparison: 
single           0.0993 MB
big_single         2.01 MB - 20.11x memory usage +1.91 MB
batched           10.03 MB - 100.43x memory usage +9.93 MB

Reduction count statistics:

Name               average  deviation         median         99th %
single              8.11 K     ±1.10%         8.06 K         8.27 K
big_single        195.82 K     ±0.14%       195.79 K       196.39 K
batched           919.38 K     ±0.91%       917.06 K       958.45 K

Comparison: 
single              8.06 K
big_single        195.82 K - 24.14x reduction count +187.71 K
batched           919.38 K - 113.33x reduction count +911.27 K
.
Finished in 17.6 seconds (0.00s async, 17.6s sync)
```

### CODE REVIEW

File: test/logflare_web/controllers/log_controller_test.exs [hunk 1]
```diff
@@ -293,4 +296,49 @@ defmodule LogflareWeb.LogControllerTest do
     reject(&Users.preload_billing_account/1)
     :ok
   end
+
+  describe "benchmarks" do
+    setup [:v1_pipeline_setup, :warm_caches, :reject_context_functions]
+
+    setup %{user: user, conn: conn} do
+      conn = put_req_header(conn, "x-api-key", user.api_key)
+
```
Comment (hauleth) [inline test/logflare_web/controllers/log_controller_test.exs]:
```suggestion
```

File: test/logflare_web/controllers/log_controller_test.exs [hunk 2]
```diff
@@ -293,4 +296,49 @@ defmodule LogflareWeb.LogControllerTest do
...
+
+      :timer.sleep(1500)
```
Comment (hauleth) [inline test/logflare_web/controllers/log_controller_test.exs]: Why it is needed?

File: test/logflare_web/controllers/log_controller_test.exs [hunk 3]
```diff
@@ -293,4 +296,49 @@ defmodule LogflareWeb.LogControllerTest do
...
+      {:ok, user: user, conn: conn}
+    end
+
+    @tag :benchmark
+    test "v1 ingestion", %{conn: conn, source: source} do
```
Comment (hauleth) [inline test/logflare_web/controllers/log_controller_test.exs:309]: Does it make sense to make it part of test suite? Wouldn't it be better to have it as a set of separate files defined somewhere?

Comment (Ziinc) [reply test/logflare_web/controllers/log_controller_test.exs:309]: it is for reusing the setups for mocks and fixtures, which lets us benchmark the larger whole of the pipeline, as opposed to small parts.

smaller functions that are isolated to a module for example can be put in a separate script.

Comment (hauleth) [reply test/logflare_web/controllers/log_controller_test.exs:309]: Hmm, ok. But I would think about [registering custom test type](https://hexdocs.pm/ex_unit/ExUnit.Case.html#register_test/4) for that case. This probably would make it a little bit easier to spot that it is not real test.

Comment (Ziinc) [reply test/logflare_web/controllers/log_controller_test.exs:309]: yea we can do that once we have a larger collection of benchmarks suites.

**Review by hauleth — APPROVED** [review]

**Pull request merged**